cgen error for `for mut ccc in aaa { match ccc {` where `aaa` is `[]string{}`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.7 62ab1be.4b4bfee
OS: linux, Ubuntu 20.04.6 LTS
Processor: 2 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/oo
vexe: /space/v/oo/v
vexe mtime: 2024-09-11 16:04:14

vroot: OK, value: /space/v/oo
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.46.0
Git vroot status: weekly.2024.36-42-g79bedff1
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg ch.v`
```v
mut aaa := []string{}
for mut ccc in aaa {
	match ccc {
		'/' {}
		else {}
	}
}

```

**What did you expect to see?**

a compiled program

**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/ch.01J7J50XKTYJ7B8WV31Q3778X0.tmp.c:13275: error: cannot convert 'struct string *' to 'struct string'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/match.v
+++ b/vlib/v/gen/c/match.v
@@ -491,7 +491,8 @@ fn (mut g Gen) match_expr_classic(node ast.MatchExpr, is_expr bool, cond_var str
 						g.write(')')
 					}
 					.string {
-						g.write('string__eq(${cond_var}, ')
+						ptr_str := if node.cond_type.is_ptr() { '*' } else { '' }
+						g.write('string__eq(${ptr_str}${cond_var}, ')
 						g.expr(expr)
 						g.write(')')
 					}

--- a/vlib/v/tests/conditions/matches/match_with_mut_cond_var_test.v
+++ b/vlib/v/tests/conditions/matches/match_with_mut_cond_var_test.v
@@ -1,0 +1,10 @@
+fn test_match_with_mut_cond_var() {
+	mut aaa := []string{}
+	for mut ccc in aaa {
+		match ccc {
+			'/' {}
+			else {}
+		}
+	}
+	assert true
+}

